### PR TITLE
Support testnet in cardano-wallet-byron:bench:restore

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#1458 

# Overview

- [x] Hard-code topology files in nix
- [x] Factor out Byron CLI parsing to be partly re-used by benchmark
- [x] Remove unused `KnownNetwork` class
- [x] Make nix set a myriad of ENV-VARS containing configuration, and let `Restore.hs` select the right ones based on either a `--mainnet` or `--testnet` flag.
- [x] Add testnet restore job to nightly.

# Comments

### CODE REVIEW

File: .buildkite/bench-restore.sh [hunk 1]
```diff
@@ -3,21 +3,28 @@
...
 echo "NODE_DB=$NODE_DB"
 
-nix-build -A benchmarks.cardano-wallet-$target.restore -o bench-$target-restore
-bench=./bench-$target-restore/bin/restore
+nix-build -A benchmarks.cardano-wallet-$target.$NETWORK-restore -o bench-$target-restore
+
+if [ "$NETWORK" = "mainnet" ]; then
+  bench="./bench-$target-restore/bin/restore"
+elif [ "$NETWORK" = "testnet" ]; then
+  bench="./bench-$target-restore/bin/restore"
```
Comment (Anviking) [inline .buildkite/bench-restore.sh]: No need for this if statement anymore.

File: nix/haskell.nix [hunk 2]
```diff
@@ -100,16 +100,26 @@ let
 
         # cardano-node will want to write logs to a subdirectory of the working directory.
         # We don't `cd $src` because of that.
-				#
-				# TODO: Using the configuration dir works for mainnet, but to add testnet
-        # support, we need to retrieve it properly.
-        packages.cardano-wallet-byron.components.benchmarks.restore =
+        packages.cardano-wallet-byron.components.benchmarks.testnet-restore =
```
Comment (Anviking) [inline nix/haskell.nix]: Some nix machinery relies on this being named `benchmarks.restore` and not `benchmarks.testnet.restore` or `benchmarks.testnet-restore`.

This fails with
```
Setup: Unknown build target 'bench:testnet-restore'.
There is no benchmark component 'testnet-restore'.
```

Not sure what to do. @rvl?

Comment (Anviking) [reply nix/haskell.nix]: I could pass in `NODE_{MAINNET, TESTNET}_{CONFIG, GENESIS}` and let the restore benchmark decide which config to use based on `--mainnet` or `--testnet` args – such that only _one_ nix wrapper is needed.

Not super-elegant, but should work.

**Comment by Anviking** [comment]
bors try

File: nix/pkgs.nix [hunk 3]
```diff
@@ -7,22 +7,24 @@ let
...
+        })
+        // { topologyFiles = {
+            # Note: It was easier to define as topologyFiles.testnet than testnet.topologyFile.
+            # TODO: Retrieve these from iohk-nix or cardano-node directly.
+            mainnet = builtins.toFile "topology" "{\"Producers\":[{\"addr\":\"relays-new.cardano-mainnet.iohk.io\",\"port\":3001,\"valency\":1}]}";
+            testnet = builtins.toFile "topology" "{\"Producers\":[{\"addr\":\"relays-new.cardano-testnet.iohkdev.io\",\"port\":3001,\"valency\":1}]}";
+          };};
 
-    # provide configuration directory as a convenience
-    configs = pkgs.runCommand "cardano-node-configs" {} ''
```
Comment (Anviking) [inline nix/pkgs.nix:23]: Ok, cannot remove this, or hydra will fail

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
I'd advocate for keeping the many ENV VARS at least until we can use relative paths in the config ( https://github.com/input-output-hk/cardano-node/pull/750).

It's just that `stack bench cardano-wallet-byron:bench:restore` is difficult to run. `./.buildkite/bench-restore.sh` and `nix-build -A benchmarks.cardano-wallet-byron.restore -o restore` are easy though.

**Comment by rvl** [comment]
Fair enough :+1:

**Review by rvl — APPROVED** [review]

**Pull request merged**